testing/internal/e2e/tests/base/paginate_credential_test.go

```diff
@@ -0,0 +1,216 @@
+			"-credential-store-id", newStoreId,
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var newCredentials credentials.CredentialListResult
+	err = json.Unmarshal(output.Stdout, &newCredentials)
+	require.NoError(t, err)
+
+	require.Len(t, newCredentials.Items, c.MaxPageSize+1)
+	// The first item should be the most recently created, which
+	// should be our new credential
+	firstItem := newCredentials.Items[0]
+	assert.Equal(t, newCredentialId, firstItem.Id)
+	assert.Empty(t, newCredentials.ResponseType)
+	assert.Empty(t, newCredentials.RemovedIds)
+	assert.Empty(t, newCredentials.ListToken)
+	// Ensure the deleted credential isn't returned
+	for _, credential := range newCredentials.Items {
+		assert.NotEqual(t, credential.Id, initialCredentials.Items[0].Id)
+	}
+}
+
+// TestApiPaginateCredentials asserts that the API automatically paginates to retrieve
+// all Credentials in a single invocation.
+func TestApiPaginateCredentials(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+	ctx := context.Background()
+	sClient := scopes.NewClient(client)
+	cClient := credentials.NewClient(client)
+	newOrgId := boundary.CreateNewOrgApi(t, ctx, client)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		_, err := sClient.Delete(ctx, newOrgId)
+		require.NoError(t, err)
+	})
+	newProjectId := boundary.CreateNewProjectApi(t, ctx, client, newOrgId)
+	newStoreId := boundary.CreateNewCredentialStoreStaticCli(t, ctx, newProjectId)
+
+	// Create enough credentials to overflow a single page.
+	var credentialIds []string
+	for i := 0; i < c.MaxPageSize+1; i++ {
+		resp, err := cClient.Create(
+			ctx,
+			"username_password",
+			newStoreId,
+			credentials.WithUsernamePasswordCredentialUsername("user"),
+			credentials.WithUsernamePasswordCredentialPassword("password"),
+		)
+		require.NoError(t, err)
+		credentialIds = append(
+			credentialIds,
+			resp.Item.Id,
+		)
+	}
+
+	// List Credentials
+	initialCredentials, err := cClient.List(ctx, newStoreId)
+	require.NoError(t, err)
+
+	var returnedIds []string
+	for _, cred := range initialCredentials.Items {
+		returnedIds = append(returnedIds, cred.Id)
+	}
+
+	require.Len(t, initialCredentials.Items, c.MaxPageSize+1)
+	assert.Empty(t, cmp.Diff(returnedIds, credentialIds, cmpopts.SortSlices(func(i, j string) bool { return i < j })))
+	assert.Equal(t, "complete", initialCredentials.ResponseType)
+	assert.Empty(t, initialCredentials.RemovedIds)
+	assert.NotEmpty(t, initialCredentials.ListToken)
+	mapItems, ok := initialCredentials.GetResponse().Map["items"]
+	require.True(t, ok)
+	mapSliceItems, ok := mapItems.([]any)
+	require.True(t, ok)
+	assert.Len(t, mapSliceItems, c.MaxPageSize+1)
+
+	// Create a new credential and destroy one of the other Credentials
+	newCredentialId := boundary.CreateNewStaticCredentialPasswordCli(t, ctx, newStoreId, "user", "password")
```

Comment:
I think we may want to use the Api method here, too?

Reply:
There isn't a helper for this. Generally, it doesn't matter to me how the setup/surrounding logic is done, as long as the method being tested is using the appropriate style (list, in this case). Does that make sense?

Reply:
Yea. I follow your point. My main concern is being able to run this test standalone if needed, so either.
- we convert this to use an Api method
- or we add in a `boundary.AuthenticateAdminCli(t, ctx)` somewhere in this test.

Reply:
Great point. I'm happy to add an API helper for this.

Reply:
Done